### PR TITLE
Use version from report as output message version

### PR DESCRIPTION
**Pull request by epapbak**
# Description

Use proper value for "version" field, now that it is included in the generated report. This also allows us to keep sending the same message as before (we keep alll the existing contracts and don't mess with the internals of our pipeline), instead of sending the report with the new "version" field included in the [new format](https://gitlab.cee.redhat.com/ccx/ccx-ocp-core/-/blob/master/ccx_ocp_core/data/rule_results_schema.json?ref_type=heads). 

If the received report does not include the version field (always expected), the output message's version will be 0, causing the aggregator to warn about improper message version.

Fixes CCXDEV-12869

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Testing steps

CI

## Checklist
* [x] `pre-commit run -a` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by JiriPapousek — APPROVED** [review]

File: ccx_messaging/publishers/rule_processing_publisher.py [hunk 1]
```diff
@@ -107,13 +106,14 @@ def publish(self, input_msg, report):
 
         try:
             msg_timestamp = input_msg["timestamp"]
+            msg_version = report.pop("version", 1)
```
Comment (epapbak) [inline ccx_messaging/publishers/rule_processing_publisher.py:109]: The "1" here is totally arbitrary. With the new formatter, the version (integer higher than 1) should always be included, so we could use 0 here to send something that the aggregator could complain about.

Comment (ikerreyes) [reply ccx_messaging/publishers/rule_processing_publisher.py:109]: if you do that, I would like to see a test to check that the default value returned is 0 (if possible)

Comment (epapbak) [reply ccx_messaging/publishers/rule_processing_publisher.py]: updated

File: ccx_messaging/publishers/rule_processing_publisher.py [hunk 2]
```diff
@@ -107,13 +106,14 @@ def publish(self, input_msg, report):
 
         try:
             msg_timestamp = input_msg["timestamp"]
+            msg_version = report.pop("version", 0)
```
Comment (joselsegura) [inline ccx_messaging/publishers/rule_processing_publisher.py:109]: Is the `"version"` something new provided by Insights engine?? IIRC this "schema version" is relative to our message format (OrgID, AccountNumber, etc), not the format of the "Report" value

Comment (epapbak) [reply ccx_messaging/publishers/rule_processing_publisher.py:109]: Coming back to this, the `version` is a new field in the insights engine's response when [using the new formatter](https://gitlab.cee.redhat.com/ccx/ccx-data-pipeline/-/merge_requests/743). But it is a field that neither we (our pipeline) nor our API's customers are interested in,

As I mentioned in this PR's description, by removing this field from the report, we keep the existing contracts (we keep providing the same response to existing smart-proxy v1 endpoints) without messing with the internals of our pipeline.

If we want to use that version as a replacement of the `outdata_schema_version` or not is open for discussion, but I think it makes sense.

Comment (joselsegura) [reply ccx_messaging/publishers/rule_processing_publisher.py:109]: Ok, it makes sense to me now.

Feel free to create a task to start ignoring the "Version" field in the consumers of this messages

**Review by joselsegura — APPROVED** [review]

**Pull request merged**